bindings/java/src/org/sleuthkit/datamodel/IngestJobInfo.java

```diff
@@ -0,0 +1,136 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.datamodel;
+
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Represents information for an ingest job.
+ */
+public final class IngestJobInfo {
+
+	private final int ingestJobId;
+	private final long dataSourceId;
+	private final String hostName;
+	private final Date startDateTime;
+	private Date endDateTime = new Date(0);
+	private final String settingsDir;
+	private final List<IngestModuleInfo> ingestModuleInfo;
+	private final SleuthkitCase skCase;
+	private IngestJobStatusType status;
+
+	/**
+	 * Constructs an IngestJobInfo that has not ended
+	 * @param ingestJobId
+	 * @param dataSourceId
+	 * @param hostName
+	 * @param startDate
+	 * @param settingsDir
+	 * @param ingestModuleInfo
+	 * @param skCase 
+	 */
+	IngestJobInfo(int ingestJobId, long dataSourceId, String hostName, Date startDateTime, String settingsDir, List<IngestModuleInfo> ingestModuleInfo, SleuthkitCase skCase) {
+		this.ingestJobId = ingestJobId;
+		this.dataSourceId = dataSourceId;
+		this.hostName = hostName;
+		this.startDateTime = startDateTime;
+		this.settingsDir = settingsDir;
+		this.skCase = skCase;
+		this.ingestModuleInfo = ingestModuleInfo;
+		this.status = IngestJobStatusType.STARTED;
+	}
+	
+	/**
+	 * Constructs an IngestJobInfo that has already ended
+	 * @param ingestJobId
+	 * @param dataSourceId
+	 * @param hostName
+	 * @param startDate
+	 * @param endDate
+	 * @param settingsDir
+	 * @param ingestModuleInfo
+	 * @param skCase 
+	 */
+	IngestJobInfo(int ingestJobId, long dataSourceId, String hostName, Date startDateTime, Date endDateTime, IngestJobStatusType status, String settingsDir, List<IngestModuleInfo> ingestModuleInfo, SleuthkitCase skCase) {
+		this.ingestJobId = ingestJobId;
+		this.dataSourceId = dataSourceId;
+		this.hostName = hostName;
+		this.startDateTime = startDateTime;
+		this.endDateTime = endDateTime;
+		this.settingsDir = settingsDir;
+		this.skCase = skCase;
+		this.ingestModuleInfo = ingestModuleInfo;
+	}
+
+	/**
+	 * @return the end date time of the job (equal to the epoch if it has not
+	 *         been set yet).
+	 */
+	public Date getEndDateTime() {
+		return endDateTime;
+	}
+
+	/**
+	 * Sets the end date for the ingest job info, and updates the database.
+	 * Cannot be done multiple times
+	 *
+	 * @param endDate the endDateTime to set
+	 *
+	 * @throws org.sleuthkit.datamodel.TskCoreException If the update fails.
+	 * @throws org.sleuthkit.datamodel.TskDataException If the job has already
+	 *                                                  been given an end date
+	 *                                                  or is not in the
+	 *                                                  database.
+	 */
```

Comment:
Please remove the "set only once" documentation and TskDataException specification from setEndDateTime